<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>780</width>
    <height>584</height>
   </rect>
  </property>
  <property name="cursor">
   <cursorShape>CrossCursor</cursorShape>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>780</width>
     <height>540</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>640</width>
     <height>480</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(40, 44, 52); 
 </string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_10">
    <property name="sizeConstraint">
     <enum>QLayout::SetMinAndMaxSize</enum>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="toolTip">
       <string/>
      </property>
      <property name="styleSheet">
       <string notr="true">color:rgb(82, 39, 255)</string>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tabWidgetPage1_2" native="true">
       <attribute name="title">
        <string>Filters</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,3,3,0,1,1">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,2,3,2,0,0">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="original_image_text">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>ORIGINAL IMAGE</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="modified_image_text">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>MODIFIED IMAGE</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="btn_go_back">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>GO BACK</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="orig_img_HLayout" stretch="3,1,3">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetMinAndMaxSize</enum>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QScrollArea" name="scroll_area_orig">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="acceptDrops">
             <bool>true</bool>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <widget class="QWidget" name="scroll_area_contents_orig_img">
             <property name="geometry">
              <rect>
               <x>145</x>
               <y>63</y>
               <width>18</width>
               <height>18</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3"/>
            </widget>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Minimum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QScrollArea" name="scroll_area_hist_orig">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <widget class="QWidget" name="scroll_area_contents_hist_orig">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>308</width>
               <height>144</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_9"/>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="filt_img_HLayout" stretch="3,1,3">
          <item>
           <widget class="QScrollArea" name="scroll_area_filt_img">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <widget class="QWidget" name="scroll_area_contents_filt_img">
             <property name="geometry">
              <rect>
               <x>145</x>
               <y>63</y>
               <width>18</width>
               <height>18</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_13"/>
            </widget>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_8">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QScrollArea" name="scroll_area_hist_filt">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <widget class="QWidget" name="scroll_area_contents_hist_filt">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>308</width>
               <height>145</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_14"/>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="filter_layout"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="2,0,0,0">
          <property name="spacing">
           <number>30</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="txt_pixel">
            <property name="cursor">
             <cursorShape>CrossCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="txt_selected_pixels_amount">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="txt_colors_avg">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="lbl_new_pixel_value">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>CHANGE PIXEL</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbl_x">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>x</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="input_pixel_x">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbl_y">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>y</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="input_pixel_y">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbl_with_value">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>WITH VALUE </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbl_r">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>R</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="input_pixel_new_r">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbl_g">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>G</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="input_pixel_new_g">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbl_y_3">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>B</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="input_pixel_new_b">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_update_pixel">
            <property name="styleSheet">
             <string notr="true">color:rgb(255, 255, 255)</string>
            </property>
            <property name="text">
             <string>CHANGE</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="OperationsTab" name="operationsTab" native="true">
       <attribute name="title">
        <string>Operations</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout" stretch=""/>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>780</width>
     <height>21</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">color:rgb(255, 255, 255);background-color: rgb(24, 24,24); 
 </string>
   </property>
   <widget class="QMenu" name="menu_image">
    <property name="styleSheet">
     <string notr="true">color:rgb(255, 255, 255);</string>
    </property>
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="btn_open"/>
    <addaction name="btn_save"/>
    <addaction name="btn_copy_img"/>
   </widget>
   <widget class="QMenu" name="menu_pixel">
    <property name="styleSheet">
     <string notr="true">color:rgb(255, 255, 255);</string>
    </property>
    <property name="title">
     <string>Pixel</string>
    </property>
    <addaction name="btn_show_pixel_value"/>
    <addaction name="btn_modify_pixel"/>
   </widget>
   <widget class="QMenu" name="menu_filter">
    <property name="title">
     <string>Filter</string>
    </property>
    <widget class="QMenu" name="btn_point_Operators">
     <property name="title">
      <string>Point Operators</string>
     </property>
     <addaction name="btn_gamma_filter"/>
     <addaction name="btn_thresholding_filter"/>
     <addaction name="btn_negative_filter"/>
    </widget>
    <widget class="QMenu" name="menuNoise">
     <property name="title">
      <string>Noise</string>
     </property>
     <addaction name="btn_gauss_noise"/>
     <addaction name="btn_rayleigh_noise"/>
     <addaction name="btn_exponential_noise"/>
     <addaction name="btn_salt_pepper_noise"/>
    </widget>
    <addaction name="btn_point_Operators"/>
    <addaction name="menuNoise"/>
   </widget>
   <addaction name="menu_image"/>
   <addaction name="menu_pixel"/>
   <addaction name="menu_filter"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="btn_open">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="btn_save">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="btn_show_pixel_value">
   <property name="text">
    <string>Show Value</string>
   </property>
  </action>
  <action name="btn_modify_pixel">
   <property name="text">
    <string>Modify Value</string>
   </property>
  </action>
  <action name="btn_copy_img">
   <property name="text">
    <string>Copy to another img</string>
   </property>
  </action>
  <action name="btn_gamma_filter">
   <property name="text">
    <string>Gamma</string>
   </property>
  </action>
  <action name="btn_thresholding_filter">
   <property name="text">
    <string>Thresholding</string>
   </property>
  </action>
  <action name="btn_negative_filter">
   <property name="text">
    <string>Negative</string>
   </property>
  </action>
  <action name="btn_gauss_noise">
   <property name="text">
    <string>Gauss</string>
   </property>
  </action>
  <action name="btn_rayleigh_noise">
   <property name="text">
    <string>Rayleigh</string>
   </property>
  </action>
  <action name="btn_exponential_noise">
   <property name="text">
    <string>Exponential</string>
   </property>
  </action>
  <action name="btn_salt_pepper_noise">
   <property name="text">
    <string>Salt and Pepper</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OperationsTab</class>
   <extends>QWidget</extends>
   <header>widgets.operations_tab.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
